sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java

```diff
@@ -211,6 +215,23 @@ public Read withTopic(String topic) {
     }
 
     /**
+     * Define the username to connect to the JMS broker (authenticated).
+     */
+    public Read withUsername(String username) {
+      checkArgument(username != null, "JmsIO.read().withUsername(username) called with null "
+          + "username");
+      return builder().setUsername(username).build();
+    }
+
+    /**
+     * Define the password to connect to the JMS broker (authenticated).
+     */
+    public Read withPassword(String password) {
```

Comment:
Make it `@Nullable` if it is allowed to be `null`?

Though I would maybe think that if someone runs this function, it should not be allowed to be `null`. They're deliberately trying to supply a password, so we should catch it if they send `null` instead.

(basically, up to you: do you want users to be able to "clear" the password here using `null`?)